Kc req changes

Citrus716 opened the pull request:
Opened by Ke Hao Chen

This pr provides a solution for alternating.py (which apparently had a practice problem but no solution in the main branch). It also fixes img_avg.py's solution to **actually** work. Lastly, it adds a bit more instruction to img_avg.py to clarify what an image (in list form) would look like.

Citrus716 committed:
Changed img_avg.py template

Added an example and other changes to improve understanding of the problem.

Citrus716 committed:
Update img_avg.py solution

Added an example and other changes to improve understanding of the problem. Didn't change the actual solution other than removing the use of functions.

Citrus716 committed:
Made the characters per line less in alternating

Made the characters per line less in practice/alternating

Citrus716 committed:
Added my solution to alternating.py.

Added my solution to alternating.py. I didn't make this problem. Good problem though.

lint-action committed:
Fix code style issues with Black

srikar-eranky reviewed (commented):
For alternating.py, you might want to convert the alternating numbers into a list, or print them all out on the same line, just to make the output look a bit cleaner. Other than that, I have no issues with these problems or solutions.

phrdang committed:
Fix style alternating.py

phrdang commented on 2_intermediate/chapter10/practice/img_avg.py:
```diff
@@ -12,27 +12,36 @@
 """
 
+# All the libraries imported
 from PIL import Image
```
@Citrus716 could you also include links to how to install all of the modules needed to run this program?

Citrus716 replied:
Done

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
 
+# All the libraries imported
 from PIL import Image
 import requests
```
@Citrus716 also be sure to include the links in the solution multi-line comment as well (just copy and paste)

Citrus716 replied:
Done

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
+"""
+for i in range(len(img)):
+    for j in range(len(img[0])):
+        x_n = [0]
```
@abhatia1205  Is there a way to make these variables more readable? At first glance, it's hard to tell exactly what `x_n` and `y_n` are supposed to be. Also, that goes the for `i` and `j` variables too - rather than explaining what they mean in a comment, could you name them in a way that makes sense just by looking at it?

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
+        x_n = [0]
+        y_n = [0]
+
+        if i == 0:
```
@abhatia1205  also, what exactly is happening here? Could you please add comments describing what the code is doing?

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
+        r_avg = -1 * img[i][j][0]
+        g_avg = -1 * img[i][j][1]
+        b_avg = -1 * img[i][j][2]
+        c = -1
```
@abhatia1205 What is `c` supposed to be for?

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
+        b_avg = -1 * img[i][j][2]
+        c = -1
+
+        for x in x_n:
```
@abhatia1205 If you could add a comment here, that would also be good to help people understand what's going on

phrdang replied:
@abhatia1205 ^^^

phrdang commented on 2_intermediate/chapter10/solutions/img_avg.py:
```diff
@@ -12,82 +12,92 @@
+
+
+# Code that displays the new image at the end.
 plt.imshow(newimg)
```
Okay so... I tried to run this solution code after I installed all the modules (yay for pip and pyenv), but I can only see the original image of the dog. I also got a `TypeError` which I'm not sure how to resolve because I don't really understand the code:
```
Traceback (most recent call last):
  File "/Users/rebeccadang/Desktop/Code/code4tomorrow/python/2_intermediate/chapter10/solutions/img_avg.py", line 115, in <module>
    plt.imshow(transpose)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/pyplot.py", line 2707, in imshow
    __ret = gca().imshow(
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/__init__.py", line 1431, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/axes/_axes.py", line 5519, in imshow
    im.set_data(X)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/image.py", line 706, in set_data
    raise TypeError("Invalid shape {} for image data"
TypeError: Invalid shape (3, 1008, 490) for image data
```

chrehall68 replied:
Outdated comment.

phrdang committed:
Fix style and clarify instructions for img_avg.py

phrdang reviewed (dismissed):
@abhatia1205 Please see my comments for what needs to be changed. Overall, I think adding more comments would be helpful and also please see the error I received (does it work on your computer? Maybe I'm doing something wrong...)

lint-action committed:
Fix code style issues with Black

phrdang committed:
Add push events to workflows

phrdang committed:
Ignore flake8 rule E203 (black is correct)

Adding the comment noqa: [error code] ignores only that line for flake8.

phrdang committed:
Remove on push event

Lintly only supports PRs

Citrus716 committed:
Added how to install libraries 

I read that PIL is dead and hasn't been updated since 2009. To get "PIL" you must pip install pillow because that's the new name. However, when you import, the code is import PIL in order to allow for backward compatibility. Anyways, I made the instructions to install the libraries.

lint-action committed:
Fix code style issues with Black

Citrus716 committed:
Added instruction to installing libraries in solution

Copy pasted from my instructions I added in practice.

Citrus716 committed:
Swapped i and j descriptions and added more.

Swapped i and j descriptions and added more. The i represents top to bottom(height) while the j represents left to right(width).

Citrus716 committed:
Merge branch 'master' into kc_req_changes

Citrus716 committed:
Merge branch 'master' into kc_req_changes

lint-action committed:
Fix code style issues with Black

Citrus716 committed:
Merge branch 'master' into kc_req_changes

lint-action committed:
Fix code style issues with Black

BenVN123 committed:
Delete this file because it no longer exists

Because this file no longer exists in the Python repo, it is throwing an error for the pull request.

BenVN123 committed:
Delete this because the file no longer exists

BenVN123 committed:
Delete this because the file no longer exists

BenVN123 committed:
Delete this because the file no longer exists

**Changes**

- Because this file no longer exists in the Python repo, it is throwing an error for the pull request.
- This file is currently in chapter 15, not 16

BenVN123 committed:
Merge pull request #126 from code4tomorrow/BenVN123-patch-14

Delete this because the file no longer exists

BenVN123 committed:
Merge pull request #125 from code4tomorrow/BenVN123-patch-9

Delete this because the file no longer exists

BenVN123 committed:
Merge pull request #123 from code4tomorrow/BenVN123-patch-2

Delete this because the file no longer exists

BenVN123 committed:
Merge pull request #124 from code4tomorrow/BenVN123-patch-4

Delete this because the file no longer exists

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Fix styling error

chrehall68 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 committed:
Merge branch 'master' into kc_req_changes

BenVN123 commented:
This PR is really old. Is it still valid?

chrehall68 commented:
Yeah so I just looked at it and it's still valid since main's img_avg solution doesn't work

chrehall68 commented:
Unfortunately, neither does this branch's img_avg solution

chrehall68 committed:
Fix it so that it works

lint-action committed:
Fix code style issues with Black

chrehall68 committed:
Fix comment so that it's w/i line length limit

chrehall68 committed:
Merge branch 'kc_req_changes' of https://github.com/code4tomorrow/python into kc_req_changes

chrehall68 commented:
Alright I fixed it. Now, it works. This is much better compared with the dysfunctional version that we have on main branch.

chrehall68 committed:
Add comments, fix small bug

The bug was that the averages were slightly off because I forgot to set y_relative_indexes to an empty list (which I now did, which is why it's now fixed)

chrehall68 committed:
Update workflows to match what it is in main branch

chrehall68 committed:
Update names.py to match what it is in main branch

BenVN123 committed:
Fix comment

BenVN123 committed:
Edit comment

BenVN123 committed:
Update alternating.py
